### DISCUSSION TITLE
IServerAddressesFeature when not using Kestrel

### DISCUSSION THREAD

**Discussion by DerekBeattieCG** [CoreWCF/CoreWCF#692] [Q&A] [answered]
I'm trying to get a CoreWCF service running in Lambda.  I was able to get it working but to do so, I had to dummy up a list of addresses in `ServiceHostObjectModel.ApplyConfiguration()`, without doing so, addresses is null.  Since Kestrel isn't setting `IServerAddressesFeature` is there a way to do that?

**Comment by DerekBeattieCG** [comment, marked as answer]
After looking at this more, the server that doesn't have the Features set is the AWS Lambda implementation of IServer, LambdaServer.  So the issue isn't with the CoreWCF impl of IServer.  If/when I get it working, I'll add an example.

**Reply by mconnew** [reply]
@birojnayak, do you have anything that you can share here? I gave Biroj a prototype of CoreWCF working in Azure functions (equivalent to AWS lambda) and Biroj was playing with the prototype to see how easy it would be to port it to lambda.

**Comment by DerekBeattieCG** [comment]
For AWS, in the class that inherits from `APIGatewayHttpApiV2ProxyFunction` I had to add the following to the `PostCreateHost` method override.

```
var server = webHost.Services.GetService<IServer>();
var serverAddressesFeature = new ServerAddressesFeature();
serverAddressesFeature.Addresses.Add("http://localhost:5000");
serverAddressesFeature.Addresses.Add("https://localhost:5001");
server!.Features.Set<IServerAddressesFeature>(serverAddressesFeature);
```
